src/main/java/uk/gov/justice/digital/hmpps/whereabouts/controllers/VideoLinkMigrationController.kt

```diff
@@ -0,0 +1,36 @@
+package uk.gov.justice.digital.hmpps.whereabouts.controllers
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.Parameter
+import io.swagger.v3.oas.annotations.tags.Tag
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.whereabouts.services.court.VideoLinkMigrationService
+
+@Tag(name = "court")
+@RestController
+@RequestMapping(value = ["court"], produces = [MediaType.APPLICATION_JSON_VALUE])
+class VideoLinkMigrationController(
+  val videoLinkMigrationService: VideoLinkMigrationService,
+) {
+  @GetMapping(
+    path = ["/video-link-bookings/migrate/{videoBookingId}"],
+    produces = [MediaType.APPLICATION_JSON_VALUE],
+  )
+  @ResponseStatus(HttpStatus.OK)
+  @Operation(
+    summary = "Returns the data for one video link booking and its associated events",
+    description = "Used to migrate the data for one video link booking",
+  )
+  @PreAuthorize("hasAnyRole('BOOK_A_VIDEO_LINK_ADMIN')")
+  fun getVideoLinkBookingToMigrate(
```

Comment:
Optional `from` and `to` date query params?

This means we can do a small test run but also do in chunks should the need arise.

Reply:
Ah - this is not THAT endpoint - this is the getOneById endpoint... called per video link booking.

Reply:
🤦